eng/pipelines/cpp-nightly.yml

```diff
@@ -0,0 +1,64 @@
+
+jobs:
+  - job:
+    displayName: Nightly vcpkg PR
+    pool:
+      name: azsdk-pool-mms-win-2019-general
+      vmImage: MMS2019
+
+    variables:
+      # The branch in the azure-sdk/vcpkg repo that is updated nigthly with the
+      # latest set of ports
+      TargetPrBranch: azure-sdk-for-cpp-nightly-verification
+
+    steps:
+      - checkout: self
+
+      - template: /eng/pipelines/templates/steps/generate-nightly-branch-name.yml
+
+      - template: /eng/pipelines/templates/steps/vcpkg-clone.yml
+        parameters:
+          Workspace: $(Pipeline.Workspace)
+          RepoOwner: azure-sdk
+          PRBranchName: $(TargetPrBranch)
+
+      - template: /eng/common/pipelines/templates/steps/set-default-branch.yml
+        parameters:
+          WorkingDirectory: $(Pipeline.Workspace)/vcpkg
+
+      # Git merge does not support the -c parameter. For the scope of this job
+      # it's ok to configure the user.name and user.email at a global level
+      - pwsh: |
+          git config --global user.name "azure-sdk" 
+          git config --global user.email "azuresdk@microsoft.com"
+        displayName: Configure git identity
+
+      - task: Powershell@2
+        inputs: 
+          filePath: eng/scripts/Update-VcpkgNightlyBranch.ps1
+          arguments: >-
+            -SourceBranchName $(NightlyBranchName)
+            -TargetBranchName $(TargetPrBranch)
+            -MainBranchName $(DefaultBranch)
+          pwsh: true
+          workingDirectory: $(Pipeline.Workspace)/vcpkg
+        displayName: Update vcpkg Nightly Branch
+
+      # Set $(HasChanges) to $true so create-pull-request.yml will push the
+      # branch.
+      - pwsh: Write-Host "##vso[task.setvariable variable=HasChanges]$true"
```

Comment:
Do you still need this now that you are resetting so the diff should happen correctly.

Reply:
This pipeline job does not reset the commit. 

There are 2 types of jobs for this process: 

* Integration -- This is the Integration stage in the unified pipeline nightly build which places the port in a nightly branch 
* Push to long-lived branch (specified in cpp-nightly.yml) -- This job runs once a day to update the long-lived branch with the latest nightly branch